mitreattack/collections/collection_to_index.py

```diff
@@ -20,11 +21,13 @@ def generate_index(name, description, root_url, files=None, folders=None, bundle
                                 argument
             folders (string[], optional): folders of collection JSON files to include in the index. Cannot be used with
                                 files argument. Will only match collections that end with a version number
-            bundles (stix bundle[], optional): array of stix bundle objects to include in the index
-            output (string, optional): filename for the generated collection index file
+            sets (meta[], optional): array of json dictionaries representing stix bundle objects or and array of
+                                     MemoryStore objects to include in the index
         """
-        if files and folders:
-            print("cannot use both files and folder at the same time, please use only one argument at a time")
+        if len([x for x in [files, folders, sets] if x]) > 1:
+            print("cannot use multiple arguments (files, folders, sets) at the same time, please use only one "
+                  "argument at a time")
```

Comment:
I love this methodology for checking if multiple input types are present! Very easy to extend if new arguments ever appeared.

Reply:
I had a bit of fun working out a list comprehension that could pull this off - it's way cleaner than the more straightforward alternative of a series of logical "AND"s.